src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -679,6 +697,43 @@ private static Expression AddResponseWritingToMethodCall(Expression methodCall,
 
             if (feature?.CanHaveBody == true)
             {
+                if (isRawBodyType)
+                {
+                    var bodyReader = httpContext.Request.BodyReader;
+
+                    while (true)
+                    {
+                        var result = await bodyReader.ReadAsync();
+                        var buffer = result.Buffer;
+
+                        if (result.IsCompleted)
+                        {
+                            if (bodyType == typeof(ReadOnlySequence<byte>))
+                            {
+                                // REVIEW: Does this need to be a copy? We can tell users to consume the buffer
+                                // immediately in the action (they can copy if they need to).
+                                return (buffer, true);
+                            }
+
+                            // This is very LOH unfriendly
+                            var copiedBuffer = buffer.ToArray();
+
+                            // Consume the whole thing
+                            bodyReader.AdvanceTo(buffer.End);
+
+                            if (bodyType == typeof(ReadOnlyMemory<byte>))
+                            {
+                                return (new ReadOnlyMemory<byte>(copiedBuffer), true);
+                            }
+
+                            return (copiedBuffer, true);
+                        }
+
+                        // Buffer the body
+                        bodyReader.AdvanceTo(buffer.Start, buffer.End);
```

Comment:
These are up for debate. It's fine for small buffers but not huge ones. We could only support `ReadOnlySequence<byte>` in a non-copying way and then force users to call `ToArray()` in their code. Another option would be to use the array pool on their behalf but I don't like that for arrays or memory.

Reply:
I decided to only support ReadOnlySequence\<byte\>